DIPs/DIP1016.md

```diff
@@ -175,6 +175,32 @@ Again where initial construction of `result` should be performed at the moment o
 
 It is important that `T` be defined as the _parameter_ type, and not `auto` (ie, the argument type), because it will allow implicit conversions to occur naturally, with identical behavior as when the parameter is not `ref`. The user should not experience edge cases, or differences in functionality when calling `fun(int x)` vs `fun(ref int x)`.
 
+### `@property` and implicit wideing casts
+
+`@property` and implicit wideing casts are syntactically identical to public feilds and variables respectively and so a change in the API
+from one to the other would result in silent change to the users code. 
+For this reason, they promote to`const ref` rather than `ref`, since this guaruntees that the caller is not expecting the value to change as a result of the call. 
+All other cases (i.e. not `@property` or implicit wideing casts) the expressions that give rise to the promoted rvalue-to-ref do not outlive the scope of the call expression so there is no risk of such use after expected modification.
+
+e.g.
+```d
+//API v1
+void foo(ref short i)
+{
+    i = 42;
+}
+//API v2
+void foo(ref int i)
+{
+    i = 42;
+}
+//user code
+short x;
+foo(x); // modify i
+```
+
+Under the change from v1 to v2 of the API the users code would otherwise continue to compile but produce unexpected results. But if the implicit promotion is to `const ref` then the code fails to compile and the user is forced to acknowledge the change in behaviour.
```

Comment:
API the users code -> API, user code/users' code

results. But if -> results. If

`const ref` then -> `const ref`, then